sklearn/neighbors/base.py

```diff
@@ -421,6 +454,174 @@ class from an array representing our data set and ask who's
                 return dist, neigh_ind
             return neigh_ind
 
+    def masked_kneighbors(self, X=None, n_neighbors=None, return_distance=True,
```

Comment:
Yes, must have been a misunderstanding. I only asked for separation in `pairwise_distances` and then only as much separation as there is between `euclidean_distances` and `cosine_distances` and `manhattan_distances`.